File: apps/backend/src/EnterpriseAgentOs.Domain/Common/ValueObjects/SessionStatus.cs
namespace EnterpriseAgentOs.Domain.Common.ValueObjects;

public enum SessionStatus
{
    Active,
    Ended,
}

public static class SessionStatusExtensions
{
    public static string ToStorageString(this SessionStatus status) => status switch
    {
        SessionStatus.Active => "active",
        SessionStatus.Ended => "ended",
        _ => throw new ArgumentOutOfRangeException(nameof(status)),
    };

    public static SessionStatus ToSessionStatus(this string value) => value switch
    {
        "active" => SessionStatus.Active,
        "ended" => SessionStatus.Ended,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown session status: {value}"),
    };
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/SessionStatus.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum SessionStatus {
    Active,
    Ended;

    public String toWire() {
        return switch (this) {
            case Active -> "active";
            case Ended -> "ended";
        };
    }

    public static SessionStatus fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "active" -> Active;
            case "ended" -> Ended;
            default -> throw new IllegalArgumentException("Unknown session status: " + value);
        };
    }
}
